### ISSUE TITLE
setting quarkus.rest-client.logging.body-limit=-1 result in stringoutofboundsexception

### ISSUE THREAD

**Issue by maxandersen** [quarkusio/quarkus#31816]
### Describe the bug

`quarkus.rest-client.logging.body-limit=0` gives you empty body, tried `quarkus.rest-client.logging.body-limit=-1` to try get unlimited but instead I get:

```
2023-03-13 17:02:40,228 ERROR [io.qua.mut.run.MutinyInfrastructure] (vert.x-eventloop-thread-2) Mutiny had to drop the following exception: java.lang.StringIndexOutOfBoundsException: begin 0, end -1, length 2830
	at java.base/java.lang.String.checkBoundsBeginEnd(String.java:4602)
	at java.base/java.lang.String.substring(String.java:2705)
	at org.jboss.resteasy.reactive.client.logging.DefaultClientLogger.bodyToString(DefaultClientLogger.java:62)
	at org.jboss.resteasy.reactive.client.logging.DefaultClientLogger.logRequest(DefaultClientLogger.java:53)
	at org.jboss.resteasy.reactive.client.handlers.ClientSendRequestHandler$2.accept(ClientSendRequestHandler.java:203)
	at org.jboss.resteasy.reactive.client.handlers.ClientSendRequestHandler$2.accept(ClientSendRequestHandler.java:114)
	at io.smallrye.context.impl.wrappers.SlowContextualConsumer.accept(SlowContextualConsumer.java:21)
	at io.smallrye.mutiny.helpers.UniCallbackSubscriber.onItem(UniCallbackSubscriber.java:73)
	at io.smallrye.mutiny.operators.uni.UniOperatorProcessor.onItem(UniOperatorProcessor.java:47)
	at io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni$UniOnItemTransformToUniProcessor.onItem(UniOnItemTransformToUni.java:60)
	at org.jboss.resteasy.reactive.client.AsyncResultUni.lambda$subscribe$1(AsyncResultUni.java:35)
	at io.vertx.core.impl.future.FutureImpl$3.onSuccess(FutureImpl.java:141)
	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
	at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)
	at io.vertx.core.impl.future.PromiseImpl.tryComplete(PromiseImpl.java:23)
	at io.vertx.core.http.impl.HttpClientImpl.lambda$null$5(HttpClientImpl.java:652)
	at io.vertx.core.impl.future.FutureImpl$3.onSuccess(FutureImpl.java:141)
	at io.vertx.core.impl.future.FutureBase.lambda$emitSuccess$0(FutureBase.java:54)
	at io.vertx.core.impl.EventLoopContext.execute(EventLoopContext.java:86)
	at io.vertx.core.impl.DuplicatedContext.execute(DuplicatedContext.java:163)
	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:51)
	at io.vertx.core.impl.future.FutureImpl.addListener(FutureImpl.java:196)
	at io.vertx.core.impl.future.PromiseImpl.addListener(PromiseImpl.java:23)
```

I would have expected a message stating it could not be negative, return everything or keep returning empty.

not a hard non graceful fail :)

**Comment by cescoffier** [comment]
It's a dropped exception. It means that the Mutiny pipeline reaches a state where the exception cannot ge passed downstream without violating the protocol, so it just logs it. Why we are in this state? That probably will answer why we have such an exception.

**Comment by maxandersen** [comment]
we are in this state because the code assumes an integer is always positive and tries to index a string with it :)

**Comment by geoand** [comment]
Asking as #31532 should have fixed it

**Comment by cescoffier** [comment]
Fixed by #31532
